botorch/optim/optimize.py

```diff
@@ -956,15 +978,20 @@ def optimize_acqf_mixed(
                     post_processing_func=post_processing_func,
                     batch_initial_conditions=batch_initial_conditions,
                     ic_generator=ic_generator,
-                    return_best_only=True,
+                    return_best_only=False,
+                    gen_candidates=gen_candidates,
+                    timeout_sec=timeout_sec / len(fixed_features_list)
+                    if timeout_sec
+                    else None,
```

Comment:
Can you move this before the `for` loop? Don't need to redo this each iteration and it will also make things more readable.